test/model/database_test.dart

```diff
@@ -174,6 +206,25 @@ void main() {
       });
       await after.close();
     });
+
+    test('upgrade to v4, with data', () async {
+      final schema = await verifier.schemaAt(3);
+      final before = v3.DatabaseAtV3(schema.newConnection());
+      await before.into(before.globalSettings).insert(
+        v3.GlobalSettingsCompanion.insert(
+          themeSetting: Value(ThemeSetting.light.name)));
+      await before.close();
+
+      final db = AppDatabase(schema.newConnection());
+      await verifier.migrateAndValidate(db, 4);
+      await db.close();
+
+      final after = v4.DatabaseAtV4(schema.newConnection());
+      final globalSettings = await after.select(after.globalSettings).getSingle();
+      check(globalSettings.themeSetting).equals(ThemeSetting.light.name);
```

Comment:
nit:
```suggestion
      check(globalSettings).themeSetting.equals(ThemeSetting.light.name);
```

(then cascade as usual)

Reply:
… Oh I see, the type here is `v4.GlobalSettingsData`, not something more widely shared. OK, no need to have a checks-extension for it.